src/sql/magic.py

```diff
@@ -619,6 +622,73 @@ def _persist_dataframe(
         display.message_success(f"Success! Persisted {table_name} to the database.")
 
 
+def find_toml_path():
+    """
+    Recursively finds an absolute path to pyproject.toml starting
+    from current directory
+    """
+    current = Path().resolve()
+    while not (current / "pyproject.toml").exists():
+        if current == current.parent:
+            return None
+
+        current = current.parent
+    display.message(f"Found pyproject.toml from '{current}'")
+
+    return str(current) + "/pyproject.toml"
+
+
+def get_configs(file_path):
+    """Returns saved configs from given file_path"""
+    with open(file_path, "r") as file:
+        content = file.read()
+        data = toml.loads(content)
+    try:
+        return data["tool"]["jupysql"]["SqlMagic"]
+    except KeyError:
+        return {}
+
+
+def get_config_variables(obj):
+    """Returns a list of configs users can set"""
+    config_vars = list(vars(obj)["_all_trait_default_generators"].keys())
+    config_vars.remove("config")
+    config_vars.remove("parent")
+    return config_vars
+
+
+def load_configs(ip):
+    """Loads saved configs in pyproject.toml"""
+    file_path = find_toml_path()
+    if file_path:
+        configs = get_configs(file_path)
+        config_vars = get_config_variables(SqlMagic)
+        for config, value in configs.items():
+            if config in config_vars:
+                default_value = getattr(SqlMagic, config).default_value
+                if isinstance(value, type(default_value)):
+                    ip.run_cell(f"%config SqlMagic.{config} = {value}")
+                    display.message(f"Setting '{config}' into '{value}'")
+                else:
+                    display.message(
+                        f"'{value}' is an invalid value for '{config}'. "
+                        f"Please use {type(default_value)} value instead."
+                    )
+            else:
+                closest_match = difflib.get_close_matches(config, config_vars, n=1)
+                if not closest_match:
+                    display.message(
+                        f"'{config}' is an invalid configuration. Please review our "
+                        "configuration guideline: "
+                        "https://jupysql.ploomber.io/en/latest/api/configuration.html#options"  # noqa breaks the check-for-broken-links
+                    )
+                else:
+                    display.message(
+                        f"'{config}' is an invalid configuration. "
+                        f"Did you mean '{closest_match[0]}'?"
```

Comment:
Would it make sense to show a few more comma separated matches instead of just the first one? I see some some similar settings `autocommit`, `autolimit`

Reply:
Made the change. The change is in `find_close_match_config`.